pkg/git/repository.go

```diff
@@ -97,6 +98,59 @@ func (r *Repository) WriteFile(src io.Reader, dst string) error {
 	return err
 }
 
+// Returns the single directory under the environments folder for a given repo
+// Returns an error if there was a problem in doing so (including if more than one folder found)
+// string return type for ease of mocking, callers would use .Name() anyway
+func (r *Repository) GetUniqueEnvironmentFolder() (string, error) {
+	topLevelDirs, err := r.DirectoriesUnderPath(r.repoPath())
+	var topLevelDirsNoDotGit []os.FileInfo
+	for _, dir := range topLevelDirs {
+		if dir.Name() != ".git" {
+			topLevelDirsNoDotGit = append(topLevelDirsNoDotGit, dir)
+			break
+		}
+	}
+	if err != nil {
+		return "", err
+	}
+	numDirs := len(topLevelDirsNoDotGit)
+	if numDirs != 1 {
+		return "", err
+	}
+	topLevelDir := topLevelDirsNoDotGit[0]
+	if topLevelDir.Name() != "environments" {
+		return "", err
+	}
+	lookup := path.Join(r.repoPath(), topLevelDir.Name())
+
+	foundDirsUnderEnv, err := r.DirectoriesUnderPath(lookup)
+	if err != nil {
+		return "", err
+	}
+	numDirsUnderEnv := len(foundDirsUnderEnv)
```

Comment:
You could combine these two lines unless you think it's more readable as it is.

Reply:
As in, combining these?

```
	numDirsUnderEnv := len(foundDirsUnderEnv) <-- this
	if numDirsUnderEnv != 1 { <-- this
		return "", fmt.Errorf("found %d directories under environments folder, wanted one. Looked under directory %s", numDirsUnderEnv, lookup)
```

If so I want the numDirsUnderEnv as a variable to come out through the error, found this useful while debugging

And if you mean combining numDirsUnderEnv with

```
	foundDirsUnderEnv, err := r.DirectoriesUnderPath(lookup)
	if err != nil {
		return "", err
	}
```

(so the above part), as that can return an error too I think it's best as is atm